src/App.tsx

```diff
@@ -14,11 +14,97 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[];
+  addedToCart: string[];
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    addedToCart: [],
+  };
+
+  addGood = (good: string) => {
+    this.setState((state) => ({
```

Comment:
тут аналогічно, назву краще змінити щоб не виникало проблем з неймінгом, бо змінна state вже оголошена

```suggestion
    this.setState((currentState) => ({
```